File: Chapter 5/5.2/kattis/trip.py
while True:
    nTerms = int(input())

    if nTerms==0:
        break

    values = []
    total = 0
    for i in range(nTerms):
        v = int(round(100*float(input()),0))
        values.append(v)
        total += v

    lower, rem = divmod(total,nTerms)
    upper = lower + 1
    
    above = 0
    below = 0
    
    for val in values:
        if val > upper:
            above += val-upper
        elif val < lower:
            below += lower-val
    
    print(f"${max(above,below)/100:.2f}")

File: Chapter 5/5.2/kattis/trip.cpp
#include<iostream>
#include<vector>
#include<string>
#include<cmath>

using namespace std;

int main(){
    string s;
    long long totalCost;
    
    vector<long long> costs {};
    while(getline(cin,s), s!= "0"){
        int nTerms = stoi(s);
        bool exact {true};

        costs.clear();
        costs.reserve(nTerms);
        totalCost = 0;

        for(int i=0;i<nTerms;++i){
            getline(cin,s);
            long long val {round(100*stold(s))};
            costs.push_back(val);
            totalCost+=val;
        }

        long long minCost {totalCost/nTerms};
        long long maxCost {minCost};
        if(totalCost%nTerms != 0){
            ++maxCost;
            exact = false;
        }

        long long above {0};
        long long below {0};

        for(long long val: costs){
            if(val >= maxCost){
                above += (val-maxCost);
                
            } else {
                below += (minCost-val);
            }
        }
        
        long long totalDiff = max(above,below);
        
        cout << "$";
        long long q {totalDiff / 100LL};
        if(q == 0){
            cout << "0.";
        } else {
            cout << q << '.';
        }

        q = totalDiff%100;
        if(q < 10){
            cout << '0';
        }
        cout << q << '\n';
    }
}
